### PR TITLE
Add support for MaxNode option with CBC solver

### PR DESCRIPTION
**Pull request by E-Aho**
This commit adds support for the `maxNodes` option for the CBC solver. Includes a simple test to make sure the option is implemented correctly.

I originally tried to create a test to show that using the `maxNodes` option would cause the CBC solver to terminate upon reaching that number of nodes, but couldn't find an intuitive way to read the `Enumerated Nodes` or "stopping reason" outputs of the sovler in test cases.

### CODE REVIEW

**Comment by E-Aho** [comment]
Pinging @pchtsp to check if there's any update on this PR :)

For reference, I've been using the change in this fork since raising the PR and it has been working fine

File: pulp/apis/core.py [hunk 1]
```diff
@@ -205,6 +205,7 @@ def __init__(
         self.msg = msg
         self.options = options
         self.timeLimit = timeLimit
+        self.maxNodes = maxNodes
```
Comment (pchtsp) [inline pulp/apis/core.py:208]: as the comment two lines below says, the maxNodes attribute is stored inside the `self.optionsDict` attribute.

File: pulp/apis/coin_api.py [hunk 2]
```diff
@@ -167,6 +170,8 @@ def solve_CBC(self, lp, use_mps=True):
             cmds += f"mips {tmpMst} "
         if self.timeLimit is not None:
             cmds += f"sec {self.timeLimit} "
+        if self.maxNodes is not None:
```
Comment (pchtsp) [inline pulp/apis/coin_api.py:173]: As the other comment says, this attribute does not exist. You should thus do something like:
```
maxNodes = self.optionsDict.get("maxNodes")
if maxNodes is not None:
    cmds += f"maxNodes {maxNodes} "
```

**Review by pchtsp — CHANGES_REQUESTED** [review]

**Comment by pchtsp** [comment]
sorry for the troubles, I've added the maxNodes argument as explained in the comments https://github.com/coin-or/pulp/pull/690. I'm closing the PR.

**Pull request closed without merging**